src/charm.py

```diff
@@ -1,33 +1,42 @@
 #!/usr/bin/env python3
-# Copyright 2020 Omnivector Solutions, LLC.
+# Copyright 2024 Omnivector, LLC.
 # See LICENSE file for licensing details.
 
-"""SlurmdCharm."""
+"""Slurmd Operator Charm."""
 
 import logging
-from pathlib import Path
+import socket
+from dataclasses import fields
+from typing import Any, Dict
 
-import distro
-from charms.fluentbit.v0.fluentbit import FluentbitClient
-from charms.operator_libs_linux.v0.juju_systemd_notices import (
+from charms.fluentbit.v0.fluentbit import FluentbitClient  # type: ignore[import-untyped]
+from charms.operator_libs_linux.v0.juju_systemd_notices import (  # type: ignore[import-untyped]
     ServiceStartedEvent,
     ServiceStoppedEvent,
     SystemdNotices,
 )
-from interface_slurmd import Slurmd
-from ops.charm import ActionEvent, CharmBase
-from ops.framework import StoredState
-from ops.main import main
-from ops.model import ActiveStatus, BlockedStatus, WaitingStatus
-from slurm_ops_manager import SlurmManager
-from utils import monkeypatch, slurmd
+from interface_slurmctld import (
+    Slurmctld,
+    SlurmctldAvailableEvent,
+)
+from ops import (
+    ActionEvent,
+    ActiveStatus,
+    BlockedStatus,
+    CharmBase,
+    ConfigChangedEvent,
+    InstallEvent,
+    RelationCreatedEvent,
+    StoredState,
+    UpdateStatusEvent,
+    WaitingStatus,
+    main,
```

Comment:
yeah ... its for `ops.main` but I see how it could be confusing ... I just felt it was cleaner to have all of the ops import together there, but I don't mind moving it to its own import line

Reply:
I'm not going to bike shed on it :rofl: 

I agree that it looks cleaner from the import point of view which I think matters more than a single line at the end of the file.